test/constrained-generics/special-method-warnings/user-enterContext-warning-generic-concrete-impl-used.chpl

```diff
@@ -0,0 +1,10 @@
+record R {
+    var x;
+}
+
+proc ref (R(int)).enterContext() ref: int {
+    return this.x;
+}
+
+var r: R(int) = new R(1);
+writeln(r.enterContext());
```

Comment:
It's probably good to have a test of the direct call, too, but I'd expect an example using it implicitly via a manage statement

Reply:
Does it affect your concern to know that this emits a different (more specific) warning, which is tested elsewhere?

```
file.chpl:1: In module 'user-contextManager-warning-generic-concrete-impl-used-stmt':
file.chpl:11: warning: R(int(64)) is being used in a 'manage' statement via its 'enterContext' and 'exitContext' methods. However, R(int(64)) does not implement contextManager. In the future, this will result in an error.
```

Reply:
I should say the warnings serve different purposes (and so aren't redundant). In the future, it won't be an error to _have_ special methods; they just won't be used. Meanwhile, it _will_ be an error to try to use those special methods in a `manage` statement, since that will require an interface.

Reply:
As long as there are already tests of it with the manage statement, I'm satisfied